bundles/framework/publisher2/tools/LayerSelectionTool.js

```diff
@@ -128,10 +137,22 @@ Oskari.clazz.define('Oskari.mapframework.publisher.tool.LayerSelectionTool',
                     var layer = layers[i];
                     me._addLayer(layer);
                 }
+                me._metadataAvailable(layers);
```

Comment:
Add `_metadataAvailable()` calls to other places where `_checkCanChangeStyle()` is called (on maplayer add/remove atleast). The init-function should also check if the checkbox is checked when editing a map. Looks like it doesn't for styleselection either, but it's done here: https://github.com/oskariorg/oskari-frontend/blob/3c333284e1c498a7d614bf085daa1dc61ae5b2c4/bundles/framework/publisher2/tools/LayerSelectionTool.js#L130-L132. We can do the same thing here.